trunk/SUAVE/Optimization/Package_Setups/particle_swarm_optimization.py

```diff
@@ -0,0 +1,169 @@
+## @ingroup Optimization-Package_Setups
+# particle_swarm_optimization.py
+# 
+# Created:  Sep. 2019, M. Clarke
+
+# ----------------------------------------------------------------------
+#  Imports
+# ----------------------------------------------------------------------
+
+# suave imports
+import numpy as np
+import scipy as sp
+ 
+## @ingroup Optimization-Package_Setups
+def particle_swarm_optimization(func, lb, ub, ieqcons=[], f_ieqcons=None, args=(), kwargs={}, 
+        swarmsize=100, omega=0.5, phip=0.5, phig=0.5, maxiter=100, 
+        minstep=1e-8, minfunc=1e-8, debug=False):
+    """
+    This function perform a particle swarm optimization (PSO)
+    
+    Source:
+        Pyswarm: https://github.com/tisimst/pyswarm
```

Comment:
Wait did you just copy their code?

Reply:
Yes, I just added a wrapper for Nexus

Reply:
Can you include information on the license?

Reply:
Says it's authorized under the BSD license which as far as I can tell only requires that we add a copyright notice and a list of conditions, available here: [3-clause BSD License](https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_(%22BSD_License_2.0%22,_%22Revised_BSD_License%22,_%22New_BSD_License%22,_or_%22Modified_BSD_License%22))

There is also a 0-clause version which is functionally a public domain license. I'd say that including the copyright notice at the bottom of the file or anywhere outside the docstring so that doxygen doesn't spit it out should suffice.

Reply:
Including that info sounds good. I'd recommend top of the file after the standard header.

Reply:
Copywrite added